### PR TITLE
Feature/add repos

### PR DESCRIPTION
**Pull request by Gh0stF1ghter**

### CODE REVIEW

File: DAL/Repositories/Implementations/ClientRepository.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+using DAL.Entities;
+using DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+
+namespace DAL.Repositories.Implementations;
+public class ClientRepository(AgencyDbContext context) : IClientRepository
+{
+    public async Task<IEnumerable<Client>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/ClientRepository.cs:11]: formatting

File: DAL/Repositories/Implementations/ManufacturerRepository.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using DAL.Context;
+using DAL.Entities;
+using DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+public class ManufacturerRepository(AgencyDbContext context) : IManufacturerRepository
+{
+    public async Task<IEnumerable<Manufacturer>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/ManufacturerRepository.cs:10]: formatting

File: DAL/Repositories/Implementations/ManufacturerRepository.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public async Task<bool> IsExistsAsync(Expression<Func<Manufacturer, bool>> predicate, CancellationToken cancellationToken) =>
+        await context.Manufacturers.AnyAsync(predicate, cancellationToken: cancellationToken);
+
+    public async Task AddAsync(Manufacturer manufacturer, CancellationToken cancellationToken)
+    {
+        await context.Manufacturers.AddAsync(manufacturer, cancellationToken);
+        await context.SaveChangesAsync(cancellationToken);
+    }
+    public async Task RemoveAsync(Manufacturer manufacturer, CancellationToken cancellationToken)
+    {
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/ManufacturerRepository.cs:35]: formatting

File: DAL/Repositories/Implementations/ModelRepository.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    public async Task<bool> IsExistsAsync(Expression<Func<Model, bool>> predicate, CancellationToken cancellationToken) =>
+        await context.Models.AnyAsync(predicate, cancellationToken);
+
+    public async Task AddAsync(Model model, CancellationToken cancellationToken)
+    {
+        await context.Models.AddAsync(model, cancellationToken);
+        await context.SaveChangesAsync(cancellationToken);
+    }
+    public async Task RemoveAsync(Model model, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/ModelRepository.cs:35]: formatting

File: DAL/Repositories/Implementations/VehicleClientHistoryRepository.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+using DAL.Entities;
+using DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+public class VehicleClientHistoryRepository(AgencyDbContext context) : IVehicleClientHistoryRepository
+{
+    public async Task<IEnumerable<VehicleClientHistory>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
+    {
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/VehicleClientHistoryRepository.cs:11]: formatting

File: DAL/Repositories/Implementations/VehicleClientHistoryRepository.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        await context.VehicleClientHistories.AnyAsync(predicate, cancellationToken);
+
+    public async Task AddAsync(VehicleClientHistory vehicleClientHistory, CancellationToken cancellationToken)
+    {
+        await context.VehicleClientHistories.AddAsync(vehicleClientHistory, cancellationToken);
+        await context.SaveChangesAsync(cancellationToken);
+    }
+    public async Task RemoveAsync(VehicleClientHistory vehicleClientHistory, CancellationToken cancellationToken)
+    {
+        context.VehicleClientHistories.Remove(vehicleClientHistory);
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/VehicleClientHistoryRepository.cs:36]: formatting

File: DAL/Repositories/Implementations/VehicleRepository.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
+﻿using DAL.Context;
+using DAL.Entities;
+using DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+public class VehicleRepository(AgencyDbContext context) : IVehicleRepository
+{
+    public async Task<IEnumerable<Vehicle>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/VehicleRepository.cs:10]: formatting

File: DAL/Repositories/Interfaces/IClientRepository.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿using DAL.Entities;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Interfaces;
+public interface IClientRepository
+{
+    public Task<IEnumerable<Client>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Client?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
+
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IClientRepository.cs:9]: formatting

File: DAL/Repositories/Interfaces/IClientRepository.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    public Task<bool> IsExistsAsync(Expression<Func<Client, bool>> predicate, CancellationToken cancellationToken);
+
+    public Task AddAsync(Client client, CancellationToken cancellationToken);
+    public Task RemoveAsync(Client client, CancellationToken cancellationToken);
+}
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IClientRepository.cs:14]: formatting

File: DAL/Repositories/Interfaces/IManufacturerRepository.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+﻿using DAL.Entities;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Interfaces;
+public interface IManufacturerRepository
+{
+    public Task<IEnumerable<Manufacturer>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Manufacturer?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
+
+    public Task<bool> IsExistsAsync(Expression<Func<Manufacturer, bool>> predicate, CancellationToken cancellationToken);
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IManufacturerRepository.cs:10]: formatting

File: DAL/Repositories/Interfaces/IModelRepository.cs [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+namespace DAL.Repositories.Interfaces;
+public interface IModelRepository
+{
+    public Task<IEnumerable<Model>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Model?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
+
+    public Task<bool> IsExistsAsync(Expression<Func<Model, bool>> predicate, CancellationToken cancellationToken);
+
+    public Task AddAsync(Model model, CancellationToken cancellationToken);
+    public Task RemoveAsync(Model model, CancellationToken cancellationToken);
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IModelRepository.cs:13]: formatting

File: DAL/Repositories/Interfaces/IVehicleClientHistoryRepository.cs [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+public interface IVehicleClientHistoryRepository
+{
+    public Task<IEnumerable<VehicleClientHistory>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<VehicleClientHistory?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
+
+    public Task<bool> IsExistsAsync(Expression<Func<VehicleClientHistory, bool>> predicate, CancellationToken cancellationToken);
+
+    public Task AddAsync(VehicleClientHistory vehicleClientHistory, CancellationToken cancellationToken);
+    public Task RemoveAsync(VehicleClientHistory vehicleClientHistory, CancellationToken cancellationToken);
+}
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IVehicleClientHistoryRepository.cs:14]: formatting

File: DAL/Repositories/Interfaces/IVehicleRepository.cs [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+public interface IVehicleRepository
+{
+    public Task<IEnumerable<Vehicle>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Vehicle?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
+
+    public Task<bool> IsExistsAsync(Expression<Func<Vehicle, bool>> predicate, CancellationToken cancellationToken);
+
+    public Task AddAsync(Vehicle vehicle, CancellationToken cancellationToken);
+    public Task RemoveAsync(Vehicle vehicle, CancellationToken cancellationToken);
+}
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IVehicleRepository.cs:14]: formatting

File: DAL/Repositories/Interfaces/IVehicleRepository.cs [hunk 14]
```diff
@@ -0,0 +1,16 @@
+﻿using DAL.Entities;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Interfaces;
+
+public interface IVehicleRepository
+{
+    public Task<IEnumerable<Vehicle>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Vehicle?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IVehicleRepository.cs:9]: formatting + you do not need to use access modifiers in interfaces

File: DAL/Repositories/Interfaces/IVehicleClientHistoryRepository.cs [hunk 15]
```diff
@@ -0,0 +1,16 @@
+﻿using DAL.Entities;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Interfaces;
+
+public interface IVehicleClientHistoryRepository
+{
+    public Task<IEnumerable<VehicleClientHistory>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<VehicleClientHistory?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
+
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IVehicleClientHistoryRepository.cs:10]: formatting + you do not need to use access modifiers in interfaces

File: DAL/Repositories/Interfaces/IModelRepository.cs [hunk 16]
```diff
@@ -0,0 +1,16 @@
+﻿using DAL.Entities;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Interfaces;
+
+public interface IModelRepository
+{
+    public Task<IEnumerable<Model>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Model?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IModelRepository.cs:9]: formatting + you do not need to use access modifiers in interfaces

File: DAL/Repositories/Interfaces/IManufacturerRepository.cs [hunk 17]
```diff
@@ -0,0 +1,16 @@
+﻿using DAL.Entities;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Interfaces;
+
+public interface IManufacturerRepository
+{
+    public Task<IEnumerable<Manufacturer>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Manufacturer?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IManufacturerRepository.cs:9]: formatting + you do not need to use access modifiers in interfaces

File: DAL/Repositories/Interfaces/IClientRepository.cs [hunk 18]
```diff
@@ -0,0 +1,16 @@
+﻿using DAL.Entities;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Interfaces;
+
+public interface IClientRepository
+{
+    public Task<IEnumerable<Client>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken);
+    public Task<Client?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
```
Comment (moltenessence) [inline DAL/Repositories/Interfaces/IClientRepository.cs:9]: formatting + you do not need to use access modifiers in interfaces

File: DAL/Repositories/Implementations/ModelRepository.cs [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+using DAL.Entities;
+using DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+
+public class ModelRepository(AgencyDbContext context) : IModelRepository
+{
+    public async Task<IEnumerable<Model>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline DAL/Repositories/Implementations/ModelRepository.cs:11]: "Model" is not the best naming for DAL level entities

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by nantonov — APPROVED** [review]

File: DAL/Repositories/Implementations/ClientRepository.cs [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+
+public class ClientRepository(AgencyDbContext context) : IClientRepository
+{
+    public async Task<IEnumerable<Client>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
+    { 
+        var rowsToSkip = page - 1 * pageSize;
```
Comment (5Ka-me) [inline DAL/Repositories/Implementations/ClientRepository.cs:13]: Is it correct expression?

File: DAL/Repositories/Implementations/ManufacturerRepository.cs [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+
+public class ManufacturerRepository(AgencyDbContext context) : IManufacturerRepository
+{
+    public async Task<IEnumerable<Manufacturer>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
+    {
+        var rowsToSkip = page - 1 * pageSize;
```
Comment (5Ka-me) [inline DAL/Repositories/Implementations/ManufacturerRepository.cs:13]: Is it correct expression?

File: DAL/Repositories/Implementations/VehicleClientHistoryRepository.cs [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+
+public class VehicleClientHistoryRepository(AgencyDbContext context) : IVehicleClientHistoryRepository
+{
+    public async Task<IEnumerable<VehicleClientHistory>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
+    {
+        var rowsToSkip = page - 1 * pageSize;
```
Comment (5Ka-me) [inline DAL/Repositories/Implementations/VehicleClientHistoryRepository.cs:13]: Is it correct expression?

File: DAL/Repositories/Implementations/VehicleModelRepository.cs [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+
+public class VehicleModelRepository(AgencyDbContext context) : IVehicleModelRepository
+{
+    public async Task<IEnumerable<VehicleModel>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
+    {
+        var rowsToSkip = page - 1 * pageSize;
```
Comment (5Ka-me) [inline DAL/Repositories/Implementations/VehicleModelRepository.cs:13]: Is it correct expression?

File: DAL/Repositories/Implementations/VehicleRepository.cs [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations;
+
+public class VehicleRepository(AgencyDbContext context) : IVehicleRepository
+{
+    public async Task<IEnumerable<Vehicle>> GetRangeAsync(int page, int pageSize, CancellationToken cancellationToken)
+    {
+        var rowsToSkip = page - 1 * pageSize;
```
Comment (5Ka-me) [inline DAL/Repositories/Implementations/VehicleRepository.cs:13]: Is it correct expression?

**Review by 5Ka-me — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

**Pull request merged**